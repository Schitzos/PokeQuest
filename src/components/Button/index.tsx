import theme from '@/theme';
import React from 'react';
import {StyleSheet, ViewStyle, Text, TouchableOpacity} from 'react-native';

interface ContentProps {
  children: React.ReactNode;
  color?: string;
  onPress?: () => void;
  size?: 'sm' | 'md' | 'lg';
  fullWidth?: boolean;
  disabled?: boolean;
}

export default function Button({
  children,
  color,
  onPress = () => null,
  size = 'md',
  fullWidth = false,
  disabled = false,
}: ContentProps) {
  const styleSize = {
    sm: {
      padding: 4,
      fontSize: 12,
      borderRadius: 4,
    },
    md: {
      padding: 8,
      fontSize: 16,
      borderRadius: 8,
    },
    lg: {
      padding: 12,
      fontSize: 16,
      borderRadius: 16,
    },
  };

  const styles = StyleSheet.create({
    base: {
      backgroundColor: color || theme.colors.primary,
      padding: styleSize[size].padding,
      borderRadius: styleSize[size].borderRadius,
      justifyContent: 'center',
      flex: fullWidth ? 1 : 0,
    },
    label: {
      fontSize: styleSize[size].fontSize,
      textAlign: 'center',
      color: theme.colors.white,
      fontFamily: theme.font.reguler,
    },
    disabled: {
      backgroundColor: theme.colors.neutral100,
    },
  });

  return (
    <TouchableOpacity
      style={[styles.base as ViewStyle, disabled && styles.disabled]}
      onPress={onPress}
      disabled={disabled}>
      <Text style={styles.label}>{children}</Text>
    </TouchableOpacity>
  );
}
