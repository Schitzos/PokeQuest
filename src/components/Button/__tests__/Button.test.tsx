import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import Button from '../index';
import '@testing-library/jest-dom';

describe('Button component', () => {
  test('matches snapshot with default props', () => {
    const {toJSON} = render(<Button onPress={() => null}>Button Text</Button>);
    expect(toJSON()).toMatchSnapshot();
  });

  test('renders correctly with default props', () => {
    const {getByText} = render(
      <Button onPress={() => null}>Button Text</Button>,
    );
    const buttonElement = getByText('Button Text');
    expect(buttonElement).toBeTruthy();
  });

  test('executes onPress function when clicked', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(<Button onPress={onPressMock}>Click Me</Button>);
    const buttonElement = getByText('Click Me');
    fireEvent.press(buttonElement);
    expect(onPressMock).toHaveBeenCalled();
  });

  test('executes default onPress function when not provided', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(<Button>Click Me</Button>);
    const buttonElement = getByText('Click Me');
    fireEvent.press(buttonElement);
    expect(onPressMock).not.toHaveBeenCalled(); // Assuming this default behavior does not call onPress
  });
});
