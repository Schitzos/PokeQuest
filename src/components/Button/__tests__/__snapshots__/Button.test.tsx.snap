// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#00AFF0",
      "borderRadius": 8,
      "flex": 0,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "PixeloidMono",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    Button Text
  </Text>
</View>
`;
